libs/extract/src/lib/dropdown/reducers.ts

```diff
@@ -318,3 +325,25 @@ export const validate = (dropdown: AbstractDropdown, validator: any) => reduce(
     }
   }
 )
+
+export const search = (dropdown: AbstractDropdown, searchInput: string ) : AbstractDropdown => reduce(
+  dropdown, {
+    options: dropdown.options.map(option => {
+      let isMatch = false;
+      const propNames = [...[dropdown.display], ...dropdown.searchableProperties||[]]
+
+      propNames.every(prop=> {
```

Comment:
The inverted isMatch logic in the every function is a bit hard to follow along. I would suggest to replace `every` with `some` and put the empty searchinput condition separately. Something like this:
```
    const isMatch =
    searchInput.length === 0 ||
    propNames.some((prop) =>
      option[prop]?.toString().toLowerCase().includes(searchInput.toLowerCase())
    );